pontus/week2/Ass3/app.js

```diff
@@ -0,0 +1,241 @@
+		}; 
+
+		this.state = { 
+			charCount: 200,
+			isError: false,
+			errorMsg: "",
+			displayChars: false
+
+		};
+
+		this.createMsg = () => { 
+			let message = this.refs.inputField.value;
+
+			// callback to wrapper
+			this.props.createMsg(this.props.listID, message);
+
+			this.refs.inputField.value = "";
+			this.refs.inputField.focus();
+		}
+	}
+
+	displayCharCount() {
+		this.setState({ displayChars: !this.state.displayChars });
+	}
+
+	charValidation() { 
+		let count = 200 - (this.refs.inputField.value.length);
+
+		if (count < 0) { 
+			this.setState({ 
+				isError: true,
+				errorMsg: "out of characters."
+			});
+		} 
+		else { 
+			this.setState({ 
+				isError: false,
+				errorMsg: ""
+			});
+		}
+
+		this.setState({ 
+			charCount : count
+		});
+	}
+
+	render() {
+		let archivedMessages = [];
+		let nonArchivedMessages = [];
+		let messages = []
+
+			for (let key in this.props.list) { 
+				if (key === 'title') continue;
+
+				if ( this.props.lists[this.props.listID][key].isArchived) {  
+					archivedMessages.push(<Message listID={this.props.listID} moveMsg={this.props.moveMsg.bind(this)} msgID={key} lists={this.props.lists} archiveMsg={this.props.archiveMsg} deleteMsg={this.props.deleteMsg} message={this.props.list[key].text}/>);
+				}
+				else { 
+					nonArchivedMessages.push(<Message listID={this.props.listID} moveMsg={this.props.moveMsg.bind(this)} msgID={key} lists={this.props.lists} archiveMsg={this.props.archiveMsg} deleteMsg={this.props.deleteMsg} message={this.props.list[key].text}/>);
+				}
+			}
+		messages = nonArchivedMessages.concat(archivedMessages);
+
+		let errorMsg =  (this.state.isError) ? { color: 'red' } : { color: 'black'};
+		let btnState = (this.state.isError) ? true : false;
+		let displayChars = (this.state.displayChars) ? {display: 'inline'} : {display: 'none' };
+
+		return (
+			<div>
+				<h2>{this.props.title}</h2>
+				<ul>
+					{messages}
+				</ul> 
+				<input onBlur={this.displayCharCount.bind(this)} onFocus={this.displayCharCount.bind(this)} onKeyUp={this.charValidation.bind(this)} ref='inputField' type="text" />
+				<button disabled={btnState} onClick={this.createMsg}>Create New Message</button>
+				<br/>
+				<div style={errorMsg}>
+					<span style={displayChars}> {this.state.charCount} </span>
+					<span>{this.state.errorMsg}</span>
+				</div>
+			</div>
+		);
+	}
+}
+NamedList.defaultProps = { title: "new list" };
+
+class Message extends React.Component { 
+	constructor(props) { 
+		super(props);
+
+		this.state = {  
+			isArchived: false,
+			showMenu: false
+		};
+	}
+
+	move(e) {
+		let currentListID = e.target.getAttribute('data-current');
+		let newListID = e.target.getAttribute('data-new');
+		let msgID = this.props.msgID;
+
+		if (currentListID == newListID) return;
+
+		// callback to wrapper
+		this.props.moveMsg(currentListID, newListID, msgID);
+	}
+
+	archOrDelMsg(e) {
+		let btnAction = e.target.getAttribute('data-action');
```

Comment:
don't fill up the DOM with stuff that doesn't belong there. You can use different functions for each action-type, that would also make it more concise and easier to understand